demos/common/cpp/utils/include/utils/ocv_common.hpp

```diff
@@ -112,51 +120,104 @@ inline void putHighlightedText(const cv::Mat& frame,
-        }
+    template<typename T>
+    void scaleCoord(T& coord) {
+        if (!doResize || scaleFactor == 1) { return; }
+        coord.x = std::floor(coord.x * scaleFactor);
+        coord.y = std::floor(coord.y * scaleFactor);
+    }
 
-        template<typename T>
-        void scaleRect(T& rect) {
-            if (!doResize || scaleFactor == 1) { return; }
-            scaleCoord(rect);
-            rect.width = std::floor(rect.width * scaleFactor);
-            rect.height = std::floor(rect.height * scaleFactor);
+    template<typename T>
+    void scaleRect(T& rect) {
+        if (!doResize || scaleFactor == 1) { return; }
+        scaleCoord(rect);
+        rect.width = std::floor(rect.width * scaleFactor);
+        rect.height = std::floor(rect.height * scaleFactor);
+    }
+
+    bool doResize;
+
+private:
+    float scaleFactor;
+    cv::Size inputSize;
+    cv::Size outputResolution;
+    cv::Size newResolution;
+};
+
+
+class InputTransform {
+public:
+    InputTransform() : reverseInputChannels(false), trivial(true) {}
+
+    InputTransform(bool reverseInputChannels, const std::string &meanValues, const std::string &scaleValues) :
+        reverseInputChannels(reverseInputChannels),
+        trivial(!reverseInputChannels && meanValues.empty() && scaleValues.empty()),
+        means(meanValues.empty() ? cv::Scalar(0.0, 0.0, 0.0) : string2Vec(meanValues)),
+        stdScales(scaleValues.empty() ? cv::Scalar(1.0, 1.0, 1.0) : string2Vec(scaleValues)) {
+    }
+
+    bool isTrivial() const {
+        return trivial;
+    }
+
+    cv::Scalar string2Vec(const std::string &string) {
+        const auto& strValues = split(string, ' ');
+        std::vector<float> values;
+        try {
+            for (auto& str : strValues)
+                values.push_back(std::stof(str));
+        } catch (const std::invalid_argument&) {
+            throw std::runtime_error("Invalid parameter --mean_values or --scale_values is provided.");
         }
+        if (values.size() != 3) {
+            throw std::runtime_error("InputTransform expects 3 values per channel, but get \"" + string + "\".");
+        }
+        return cv::Scalar(values[0], values[1], values[2]);
+    }
 
-        bool doResize;
+    void setPrecision(const InferenceEngine::InputInfo::Ptr& input) {
+        const auto precision = isTrivial() ? InferenceEngine::Precision::U8 : InferenceEngine::Precision::FP32;
+        input->setPrecision(precision);
+    }
+
+    cv::Mat operator()(const cv::Mat& inputs) {
+        if (isTrivial()) { return inputs; }
+        cv::Mat result;
+        inputs.convertTo(result, CV_32F);
+        if (reverseInputChannels) {
+            cv::cvtColor(result, result, cv::COLOR_BGR2RGB);
+        }
+        return (result - means) / stdScales;
+    }
 
-    private:
-        float scaleFactor;
-        cv::Size inputSize;
-        cv::Size outputResolution;
-        cv::Size newResolution;
+private:
+    bool reverseInputChannels;
+    bool trivial;
```

Comment:
NIT (not necessary change): bool variables names should start from is or should (to reflect that there's true or false value inside) so I suggest "isTrivial"

Reply:
I agree, but we cannot have `bool isTrivial` and `bool isTrivial() const` at the same time, so I renamed it.

Reply:
that's fine for getter function (if they share same name). But I won't insist :)

Reply:
I removed the public getter, because now it's not used anywhere (and `bool isTrivial` shouldn't be used outside `InputTransform`)